homeassistant/components/hydrawise/sensor.py

```diff
@@ -75,3 +140,32 @@ def _update_attrs(self) -> None:
                 self._attr_native_value = dt_util.as_utc(next_run.start_time)
             else:
                 self._attr_native_value = datetime.max.replace(tzinfo=dt_util.UTC)
+        elif self.entity_description.key == "daily_active_water_use":
```

Comment:
This is getting a bit unwieldy. I think we should make a method for each sensor type that returns the native value for that sensor type. Then the conditional here can either just be a conditional with `self._attr_native_value` assignments, dynamically dispatch to the methods with something like `getattr(self, f'_get_{self.entity_description.key')` or via a manual dict mapping like `return {"daily_active_water_use": self._get_daily_active_water_use, ...}[self.entity_description.key]()`

That at least reduces so much nesting of this logic.

Reply:
Good idea. Updated.